apis/serving/v1beta1/inferenceservice_types.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2021 IBM Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//+kubebuilder:skip
+package v1beta1
+
+import (
+	"github.com/kserve/modelmesh-serving/apis/serving/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	SecretKeyAnnotation      = "serving.kserve.io/secretKey"
+	DeploymentModeAnnotation = "serving.kserve.io/deploymentMode"
+	SchemaPathAnnotation     = "serving.kserve.io/schemaPath"
+	MMDeploymentModeVal      = "ModelMesh"
+)
+
+// InferenceServiceSpec defines the desired state of InferenceService
+type InferenceServiceSpec struct {
+	// Predictor defines the model serving spec
+	// +required
+	Predictor InferenceServicePredictorSpec `json:"predictor"`
+}
+
+// InferenceServiceStatus defines the observed state of InferenceService
+type InferenceServiceStatus struct {
+	URL string `json:"url,omitempty"`
+
+	v1alpha1.PredictorStatus `json:",inline"`
+
+	// Conditions the latest available observations of a resource's current state.
+	Conditions Conditions `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+type Conditions []Condition
+
+// This is used for reflecting the Ready status of the resource
+// according to the InferenceService CRD's expected key which is:
+// status.conditions[?(@.type=='Ready')].status
+type Condition struct {
+	Type string `json:"type"`
+
+	Status corev1.ConditionStatus `json:"status"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// InferenceService is the Schema for the inferenceservices API
+type InferenceService struct {
```

Comment:
Is there any reason why we can't use the type from kserve and import the go module?

Reply:
I recall trying this before and running into some go dependency issues. This was just easier for me at this current stage.